src/controllers/transaction/index.ts

```diff
@@ -25,9 +25,25 @@ export class TransactionController extends Controller {
    * @Query apiType lists all transactions by that type
    */
   @Response<BadRequest>(400, 'Request was invalid')
+  @Response<NotFound>(404, 'Item not found')
   @Get('/')
-  public async getAllTransactions(@Query() apiType?: TransactionApiType): Promise<TransactionResponse[]> {
-    if (apiType) return await this.db.getTransactionsByType(apiType)
+  public async getAllTransactions(
+    @Query() apiType?: TransactionApiType,
+    @Query() status?: TransactionState
+  ): Promise<TransactionResponse[]> {
+    if (apiType && apiType != undefined && !status) return await this.db.getTransactionsByType(apiType)
+
+    if (!apiType && status && status != undefined) return await this.db.getTransactionsByState(status)
+
+    // Checks if both are truthy and that both are valid in the enum
+    // Status has to be checked to stop 'Type 'undefined' is not assignable to type 'TransactionState' error
+    if (
+      apiType &&
+      status &&
+      Object.values(TransactionApiType).includes(apiType as TransactionApiType) &&
+      Object.values(TransactionState).includes(status as TransactionState)
+    )
+      return await this.db.getTransactionsByStateAndType(status, apiType)
```

Comment:
rather than checking `apiType !=undefined` and `status !=undefined` etc. I think you might be able to always pass both by using optional parameters for `getTransactionsByStateAndType`. If it's not passed a value use `*` (meaning any) for the where clause. e.g. if `status=submitted` and `apiType=undefined`, then the db query is `where({ state: 'submitted', api_type: '*' }).`

Reply:
and it would mean the only db query needed is `getTransactionsByStateAndType`